cpp/inc/bond/core/detail/alloc.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <bond/core/config.h>
+
+#include <boost/utility/enable_if.hpp>
+
+#include <type_traits>
+
+
+namespace bond
+{
+namespace detail
+{
+
+/// @brief Helper type that holds an allocator.
+///
+/// @remarks Applies empty-base-optimization when possible.
+template <typename Alloc, typename Enable = void>
+class allocator_holder;
+
+template <typename Alloc>
+using empty_base_eligible = std::integral_constant<bool,
+       std::is_empty<Alloc>::value
+    && std::is_copy_constructible<Alloc>::value
+#if __cplusplus >= 201402L
```

Comment:
Make this the first condition to avoid having `>` on a single line (31)? :)

Reply:
I tried a few different ways to lay this out. The cleanest I found is adding a vacuous `true`.